### ISSUE TITLE
botbuilder-ai npm issue

### ISSUE THREAD

**Issue by tmuras** [Microsoft/botbuilder-js#590]
The Azure deployment fails:
```
\> npm i botbuilder-ai
[37;40mnpm [31;40mERR! tar.unpack untar error D:\local\Temp\monacositeextension\temp\botbuilder-ai\4.0.8\package.tgz
[37;40mnpm [30;43mWARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.2 (node_modules\chokidar\node_modules\fsevents):
[37;40mnpm [30;43mWARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
[37;40mnpm [30;43mWARN echobot-with-counter@1.0.0 No repository field.
[37;40mnpm [31;40mERR! Windows_NT 10.0.14393
[37;40mnpm [31;40mERR! argv "D:\\Program Files (x86)\\nodejs\\6.9.1\\node.exe" "D:\\Program Files (x86)\\npm\\3.10.8\\node_modules\\npm\\bin\\npm-cli.js" "i" "botbuilder-ai"
[37;40mnpm [31;40mERR! node v6.9.1
[37;40mnpm [31;40mERR! npm  v3.10.8
[37;40mnpm [31;40mERR! path D:\home\site\wwwroot\node_modules\.staging\botbuilder-ai-3c360c3a
[37;40mnpm [31;40mERR! code EEXIST
[37;40mnpm [31;40mERR! errno -4075
[37;40mnpm [31;40mERR! syscall mkdir

[37;40mnpm [31;40mERR! EEXIST: file already exists, mkdir 'D:\home\site\wwwroot\node_modules\.staging\botbuilder-ai-3c360c3a'
[37;40mnpm [31;40mERR! File exists: D:\home\site\wwwroot\node_modules\.staging\botbuilder-ai-3c360c3a
[37;40mnpm [31;40mERR! Move it away, and try again.

[37;40mnpm [31;40mERR! Please include the following file with any support request:
[37;40mnpm [31;40mERR!     D:\home\site\wwwroot\npm-debug.log
```

On linux machine it works with a warning:

```
➜ npm i botbuilder-ai
npm WARN tar ENOTEMPTY: directory not empty, rmdir '/tmp/bot/node_modules/.staging/botbuilder-ai-8a3ababe'
npm WARN tar ENOTEMPTY: directory not empty, rmdir '/tmp/bot/node_modules/.staging/botbuilder-ai-8a3ababe'
npm WARN saveError ENOENT: no such file or directory, open '/tmp/bot/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/tmp/bot/package.json'
npm WARN bot No description
npm WARN bot No repository field.
npm WARN bot No README data
npm WARN bot No license field.

+ botbuilder-ai@4.0.8
added 174 packages from 230 contributors and audited 685 packages in 13.159s
found 1 moderate severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
```

[bug]

**Comment by dmvtech** [comment]
That is a somewhat older version of npm. Can you please try updating npm and then trying again?

**Comment by tmuras** [comment]
This is from web bot installed in Azure portal - I can't update node there. Maybe it is an issue for the Azure infrastructure team.

**Comment by dmvtech** [comment]
Gotcha. Sorry about that. Can you give me more information on the exact steps you are doing to see if I can reproduce it? 

By going into my bot application, I can open the console and check that npm version is 5.6, although I'm not able to run the command from there.
![image](https://user-images.githubusercontent.com/43397658/47869809-1f67eb00-ddc5-11e8-95d1-bd5df2bc1b7c.png)

**Comment by tmuras** [comment]
Remove the dash (-) before "i". Try that:

`npm i botbuilder-ai`

**Comment by dmvtech** [comment]
Doh, of course! Thank you.

Doing that, I get some other/different exceptions. I've also noticed that I can get quite a range of versions, from web app to web app. All the way from 1.4.28 to 5.6.0.

I found the following [page ](https://blogs.msdn.microsoft.com/azureossds/2016/04/20/nodejs-and-npm-versions-on-azure-app-services/) which suggests adding a key/value pair to app settings : WEBSITE_NPM_DEFAULT_VERSION=5.6.0. But I couldn't get that to work.

So, apart from all that; is there a need behind using this approach? It is not something I have seen before. Typically the packages are installed to a local machine location, used along with your development environment/code editor and then built and published to the web application. I would suggest going this route if possible, as it is well proven. If there is a true need to go about it in the way you are, let me know and we can see if we can figure something out.
